gitflow_toolbox/common/gitlab.py

```diff
@@ -10,15 +10,11 @@
 class CurrentGitlab(gitlab.Gitlab, metaclass=Singleton):
 
     __project = None
+    private_token = None
 
     def __init__(self):
-        # If running in a gitlab CI job
-        if os.getenv("GITLAB_CI"):
-            # Use provided token, if not available, use default token from job
-            private_token = get_env("GITLAB_PRIVATE_TOKEN", "CI_JOB_TOKEN")
-            super().__init__(get_env("CI_SERVER_URL"), private_token)
-        else:
-            super().__init__(get_env("GITLAB_URL"), get_env("GITLAB_PRIVATE_TOKEN"))
+        self.private_token = get_env("GITLAB_PRIVATE_TOKEN", "CI_JOB_TOKEN")
```

Comment:
```suggestion
        self.__private_token = get_env("GITLAB_PRIVATE_TOKEN", "CI_JOB_TOKEN")
```

Reply:
`private_token` attribute is used to generate a pre-authenticated url for `push` (push.py) and so needs to be accessible

Reply:
Maybe we can refactor the authenticated URL generation into a class method :)

Reply:
No, it's ok for me